File: Python/convert-a-number-to-hexadecimal.py
class Solution:
    def toHex(self, num):
        """
        :type num: int
        :rtype: str
        """
        if num == 0:
            return '0'

        m = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'a', 'b', 'c', 'd', 'e', 'f']
        res = ''
        i = 0

        while num != 0 and i < 8:
            res = m[num & 0b1111] + res
            num >>= 4
            i += 1

        return res

File: Swift/convert-a-number-to-hexadecimal.swift
class Solution {
    func toHex(_ num: Int) -> String {
        if num == 0 {
            return "0"
        }

        let map = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "a", "b", "c", "d", "e", "f"]
        var num = num
        var res = ""
        var i = 0

        while num != 0 && i < 8 {
            res = map[num & 0b1111] + res
            num >>= 4
            i += 1
        }

        return res
    }
}
